### PR TITLE
add get_num_errors() and get_num_warnings()

### PR DESCRIPTION
**Pull request by Qining**

### CODE REVIEW

File: libshaderc/src/shaderc_test.cc [hunk 1]
```diff
@@ -418,6 +431,26 @@ TEST_F(CompileStringWithOptionsTest, GenerateDebugInfoDisassembly) {
...
 }
 
+TEST_F(CompileStringWithOptionsTest, GetNumErrors) {
+  shaderc_spv_module_t result =
+      Compilation(compiler_.get_compiler_handle(), kTwoErrorsShader,
+                  shaderc_glsl_vertex_shader)
+          .result();
+  // Expects compilation failure and two errors.
+  EXPECT_FALSE(shaderc_module_get_success(result));
+  EXPECT_EQ(2, shaderc_module_get_num_errors(result));
```
Comment (dneto0) [inline libshaderc/src/shaderc_test.cc]: I'd also check that num warnings is 0.

File: libshaderc/src/shaderc_test.cc [hunk 2]
```diff
@@ -418,6 +431,26 @@ TEST_F(CompileStringWithOptionsTest, GenerateDebugInfoDisassembly) {
...
+}
+
+TEST_F(CompileStringWithOptionsTest, GetNumWarnings) {
+  shaderc_spv_module_t result =
+      Compilation(compiler_.get_compiler_handle(), kTwoWarnings,
+                  shaderc_glsl_vertex_shader)
+          .result();
+  // Expects compilation success with two warnings.
+  EXPECT_TRUE(shaderc_module_get_success(result));
```
Comment (dneto0) [inline libshaderc/src/shaderc_test.cc]: I'd also check that num errors is 0.

**Comment by dneto0** [comment]
Also, should check that a regular compile has 0 warnings and errors.

**Comment by dneto0** [comment]
-1 please fix the comment in the test.

File: libshaderc/include/shaderc.hpp [hunk 3]
```diff
@@ -67,6 +67,22 @@ class SpvModule {
     return shaderc_module_get_length(module_);
   }
 
+  // Returns the number of warnings generated during the compilation.
+  size_t GetNumWarnings() const {
```
Comment (dneto0) [inline libshaderc/include/shaderc.hpp]: Perhaps for another CL, we should document (and test) the behaviour of the SpvModule getters when no compiled module exists.

**Comment by Qining** [comment]
Already merged

**Pull request closed without merging**